tests/Hateoas/Tests/HateoasBuilderTest.php

```diff
@@ -118,14 +118,52 @@ public function testCyclicalReferences()
                 .'"name":"reference1",'
                 .'"_embedded":{'
                     .'"reference2":{'
-                        .'"name":"reference2",'
-                        .'"_embedded":{'
-                            .'"reference1":null'
-                        .'}'
+                        .'"name":"reference2",'.
+                        '"_embedded":{'.
+                            '"reference1":null'.
+                        '}'
                     .'}'
                 .'}'
             .'}',
-            $hateoas->serialize($reference1, 'json')
+            $hateoas->serialize($reference1, 'json', SerializationContext::create()->setSerializeNull(true))
+        );
+    }
+
+    public function testShouldNotSerialiseNull()
+    {
+        $hateoas = HateoasBuilder::create()->build();
+
+        $reference1 = new CircularReference1();
```

Comment:
I think this test should not use circular references but simple objects, where one of them is null... isnt?

Reply:
I don't think so since 1 is using 2 and 2 is using 1 so it a circular reference isn't it ?

Reply:
Correct and it's also why the reference1 is serialised do you think ? Since I am not able to rejext the  <entry rel="reference1"/> from the XMl response

if you have an idea would be appreciate

Reply:
Ok I changed the test name